addons/stock/models/stock_move.py

```diff
@@ -605,11 +605,42 @@ def write(self, vals):
             self._propagate_product_packaging(vals['product_packaging_id'])
         if 'date_deadline' in vals:
             self._set_date_deadline(vals.get('date_deadline'))
+        if 'product_id' in vals:
+            orderpoint_domain = [('product_id', 'in', self.product_id.ids)]
+            wh_ids = self.location_id.warehouse_id.ids + self.location_dest_id.warehouse_id.ids
+            if wh_ids:
+                orderpoint_domain.append(('warehouse_id', 'in', wh_ids))
+            self.env.add_to_compute(
+                self.env['stock.warehouse.orderpoint']._fields['qty_to_order'],
+                self.env['stock.warehouse.orderpoint'].search(orderpoint_domain)
```

Comment:
Force the order with `order='id'`, because the `_order` of `stock.warehouse.orderpoint` isn't trivial and generate LEFT JOIN in the query.

Reply:
Should you add `sudo()` before searching (we do it in the `modified`) ?

Reply:
Yes, that's better, it also makes it possible to drop the sudo in purchase_requisition :+1: